avm/res/network/vpn-server-configuration/tests/e2e/waf-aligned/main.test.bicep

```diff
@@ -0,0 +1,87 @@
+targetScope = 'subscription'
+
+metadata name = 'WAF-aligned'
+metadata description = 'This instance deploys the module in alignment with the best-practices of the Azure Well-Architected Framework.'
+
+// ========== //
+// Parameters //
+// ========== //
+
+@description('Optional. The name of the resource group to deploy for testing purposes.')
+@maxLength(90)
+// e.g., for a module 'network/private-endpoint' you could use 'dep-dev-network.privateendpoints-${serviceShort}-rg'
+param resourceGroupName string = 'dep-${namePrefix}-network.vpnserverconfiguration-${serviceShort}-rg'
+
+@description('Optional. The location to deploy resources to.')
+param resourceLocation string = deployment().location
+
+@description('Optional. A short identifier for the kind of deployment. Should be kept short to not run into resource-name length-constraints.')
+// e.g., for a module 'network/private-endpoint' you could use 'npe' as a prefix and then 'waf' as a suffix for the waf-aligned test
+param serviceShort string = 'vscwaf'
+
+@description('Optional. A token to inject into the name of each resource. This value can be automatically injected by the CI.')
+param namePrefix string = '#_namePrefix_#'
+
+
+// ============ //
+// Dependencies //
+// ============ //
+
+// General resources
+// =================
+resource resourceGroup 'Microsoft.Resources/resourceGroups@2021-04-01' = {
+  name: resourceGroupName
+  location: resourceLocation
+}
+
+module nestedDependencies 'dependencies.bicep' = {
+  scope: resourceGroup
+  name: '${uniqueString(deployment().name, resourceLocation)}-nestedDependencies'
+  params: {
+    virtualWANName: 'dep-${namePrefix}-vw-${serviceShort}'
+    location: resourceLocation
+  }
+}
+
+// ============== //
+// Test Execution //
+// ============== //
+
+@batchSize(1)
+module testDeployment '../../../main.bicep' = [
+  for iteration in ['init', 'idem']: {
+    scope: resourceGroup
+    name: '${uniqueString(deployment().name, resourceLocation)}-test-${serviceShort}-${iteration}'
+    params: {
+      location: resourceLocation
+      name: '${namePrefix}${serviceShort}VPNConfig'
+      aadAudience: '11111111-1234-4321-1234-111111111111'
+      aadIssuer: 'https://sts.windows.net/11111111-1111-1111-1111-111111111111/'
+      aadTenant: 'https://login.microsoftonline.com/11111111-1111-1111-1111-111111111111'
+      p2sConfigurationPolicyGroups: [
+        {
+          userVPNPolicyGroupName: 'DefaultGroup'
+          policymembers: [
+            {
+            name: 'UserGroup1'
+            attributeType: 'AADGroupId'
+            attributeValue: '11111111-1111-2222-3333-111111111111'
+            }
+          ]
+          priority: '0'
+          isDefault: 'true'
+        }
+      ]
+      vpnAuthenticationTypes: [
+        'AAD'
+      ]
+      vpnProtocols: [
+        'OpenVPN'
+      ]
+      lock: {
+        name: 'myCustomLockName'
+        kind: 'CanNotDelete'
+      }
```

Comment:
```suggestion
```

not required for WAF-alignement. Please note that this change requires the readme to be regenerated